math/convertall/Portfile

```
@@ -0,0 +1,44 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           python 1.0
+PortGroup           qt5 1.0
+
+github.setup        doug-101 ConvertAll 0.8.0 v
+github.tarball_from releases
+
+python.versions     34 35 36 37 38 39
+python.default_version     39
+
+qt5.min_version     5.4
+
+name                convertall
+version             0.8.0
+license             GPL-2+
+categories          math science
+platforms           darwin
+maintainers         {@khorton kilohotel.com:kevin01} openmaintainer
+description         Extremely flexible unit converter
+long_description    ConvertAll has a large database of units, and allows conversions \
+                    that use multiple units, e.g. convert from feet per decade to \
+                    nautical miles per fortnight.
+homepage            http://convertall.bellz.org
+
+checksums           rmd160  89fcc2aa9bad7ecc1de7bee4edea68fdff39706a \
+                    sha256  624c8a792b0bc7ff3776499c2c743b32273569efd0567615e570a7e739e8d521 \
+                    size    281055
+
+depends_lib-append  port:py${python.version}-pyqt5
+
+use_configure       no
+build.cmd           ${python.bin} install.py
+build.args          -p ${prefix} \
+                    -b ${destroot}
+
+destroot {}
+
+post-patch {
+    reinplace "s|/usr/local|${prefix}|g" ${worksrcpath}/install.py
+    reinplace "s|buildRoot = '/'|buildRoot = '${workpath}/destroot'|g" ${worksrcpath}/install.py
```

I think this would make more sense as a [patch](https://guide.macports.org/#development.patches.source), since you're only replacing two constants.

Let me know if you need any help with this, and I'd be happy to assist :)

---

I've reviewed the MacPorts docs related to patches, and looked at a few existing ports, but I must admit I'm not yet understanding the advantage of a patch over using a post-patch reinplace here.  If I understand correctly, I shouldn't hard code /opt/local into a patch.  The patch would need to replace the existing /usr/local with some placeholder, then I would need a post-patch reinplace to replace that placeholder with ${prefix}.  Or, is MacPorts smart enough to replace a hard coded /opt/local in a patch with the correct PREFIX?

I must be missing something here.  Please educate me.

---

> If I understand correctly, I shouldn't hard code /opt/local into a patch. The patch would need to replace the existing /usr/local with some placeholder, then I would need a post-patch reinplace to replace that placeholder with ${prefix}.

Sorry about that @khorton, you're completely right. I didn't think it through properly, so that mistake is on me :)

---

> If I understand correctly, I shouldn't hard code /opt/local into a patch. The patch would need to replace the existing /usr/local with some placeholder, then I would need a post-patch reinplace to replace that placeholder with ${prefix}.

Yes, that is a pattern you will see used by many ports.

P.S: Use `${destroot}` here, not `${workpath}/destroot`.

```suggestion
    reinplace "s|/usr/local|${prefix}|g" ${worksrcpath}/install.py
    reinplace "s|buildRoot = '/'|buildRoot = '${destroot}'|g" ${worksrcpath}/install.py
```